src/handle_action.cpp

```diff
@@ -617,11 +617,29 @@ static void smash()
 {
     player &u = g->u;
     map &m = g->m;
-
+    if( u.has_effect( effect_riding ) && u.mounted_creature ) {
+        auto mons = u.mounted_creature.get();
+        if( mons->has_flag( MF_RIDEABLE_MECH ) ) {
+            if( !mons->check_mech_powered() ) {
+                add_msg( m_bad, _( "Your %s refuses to move as its batteries have been drained." ),
+                         mons->get_name() );
+                return;
+            }
+        }
+    }
     const int move_cost = !u.is_armed() ? 80 : u.weapon.attack_time() * 0.8;
     bool didit = false;
+    bool mech_smash = false;
+    int smashskill;
     ///\EFFECT_STR increases smashing capability
-    int smashskill = u.str_cur + u.weapon.damage_melee( DT_BASH );
+    if( u.has_effect( effect_riding ) && u.mounted_creature ) {
+        auto mon = u.mounted_creature.get();
+        smashskill = u.str_cur + mon->type->mech_str_bonus + ( mon->type->melee_dice *
+                     mon->type->melee_sides );
```

Comment:
The brackets `()` around the multiplication are not needed. `a + b * c` is always interpreted as `a + (b*c)` by the compiler.